jbpm-services/jbpm-kie-services/src/main/java/org/jbpm/kie/services/impl/RuntimeDataServiceImpl.java

```diff
@@ -809,7 +819,7 @@ public UserTaskInstanceDesc getTaskById(Long taskId) {
 	    
 	    Map<String, Object> params = new HashMap<String, Object>();
         params.put("userId", userId);
-        params.put("groupIds", getAuthenticatedUserRoles(identityProvider));
+        params.put("groupIds", userGroupCallback.getGroupsForUser(userId));
```

Comment:
@tsurdilo take a look at org.jbpm.kie.services.impl.CommonUtils.getAuthenticatedUserRoles(IdentityProvider) that does some additional logic in case roles are empty list. There is an issue with some dbs handling empty list as part of in clause